### PR TITLE
chown: fix #2060

### LINKED ISSUE #2060: chown <user>: file is not working
I recently noticed that the command `chown` is not working correctly.
When executing `chown <user>: file.txt` the command succeeds without any error, but not the user nor the group is changed.

I think this behaviour should be covered by the test `test_chown_myself_second`, but there only the succeed is checked, not the change of the user.

I sadly don't know how this could be checked in the tests nor how the group is determined just from the username :cry:. Maybe somebody else could solve this.

### PR DESCRIPTION
**Pull request by jhscheer**
fixes #2060

### CODE REVIEW

File: tests/by-util/test_chown.rs [hunk 1]
```diff
@@ -77,29 +77,42 @@ fn test_chown_myself() {
...
-        .run();
+        .succeeds()
+        .stderr_contains(&"retained as");
 
-    println!("result.stdout = {}", result.stdout);
-    println!("result.stderr = {}", result.stderr);
-    assert!(result.success);
+    // try to change to another user,
+    // this fails only if username is recognized correctly
+    let result = scene
```
Comment (sylvestre) [inline tests/by-util/test_chown.rs:109]: please fix the warning here

Comment (jhscheer) [reply tests/by-util/test_chown.rs]: done

**Comment by sylvestre** [comment]
https://github.com/uutils/coreutils/pull/2063/checks?check_run_id=2311940222 is failing

**Comment by jhscheer** [comment]
> https://github.com/uutils/coreutils/pull/2063/checks?check_run_id=2311940222 is failing

The failed tests don't seem to be related to the change in this PR.
For me it looks like `whoami` is doing some shenanigans on the testing environment (`whoami` doesn't return anything and `chown` is called with `:` only).
This is tricky to debug since I don't have failed tests on my development boxes.

The tests in `chown` need to be [refactored](https://github.com/uutils/coreutils/issues/1982#issue-846625846) anyway.
I will do that and write some more exception handling for the `whoami` calls and see if it changes anything.

**Comment by sylvestre** [comment]
I retriggered the tests

**Comment by jhscheer** [comment]
This needs #2068 to get merged first.

**Comment by jhscheer** [comment]
Should be okay now.

(there's a failed check on macos not coming from `test_chown` but from `test_cp::test_cp_parents`, which looks random to me)

**Pull request merged**